src/core/geometry/geom3/union.js

```diff
@@ -20,35 +20,34 @@ const unionForNonIntersecting = (otherGeom3, csg) => {
 const unionSub = (otherGeom3, csg, doRetesselate, doCanonicalize) => {
   if (!isOverlapping(otherGeom3, csg)) {
     return unionForNonIntersecting(otherGeom3, csg)
-  } else {
-    let a = new Tree(otherGeom3.polygons)
-    let b = new Tree(csg.polygons)
-    a.clipTo(b, false)
+  }
+  const a = new Tree(otherGeom3.polygons)
+  const b = new Tree(csg.polygons)
+  a.clipTo(b, false)
 
-    // b.clipTo(a, true); // ERROR: this doesn't work
-    b.clipTo(a)
-    b.invert()
-    b.clipTo(a)
-    b.invert()
+  // b.clipTo(a, true); // ERROR: this doesn't work
+  b.clipTo(a)
+  b.invert()
+  b.clipTo(a)
+  b.invert()
 
-    let newpolygons = a.allPolygons().concat(b.allPolygons())
-    let result = fromPolygons(newpolygons)
-    // FIXME: what to do with properties ????
-    // result.properties = otherGeom3.properties._merge(csg.properties)
-    if (doRetesselate) result = retessellate(result)
-    if (doCanonicalize) result = canonicalize(result)
-    return result
-  }
+  const newpolygons = a.allPolygons().concat(b.allPolygons())
+  let result = fromPolygons(newpolygons)
+  // FIXME: what to do with properties ????
+  // result.properties = otherGeom3.properties._merge(csg.properties)
+  if (doCanonicalize) result = canonicalize(result)
+  if (doRetesselate) result = retessellate(result)
+  return result
 }
 
 const union = (...solids) => {
-  let csgs = solids
+  const csgs = solids.slice()
```

Comment:
this should be done like the other boolean operations (arrays etc)

Reply:
I added tests for zero or one geometry passed to difference, intersection, and union. All behave the same now, and return undefined when passed zero geomeries, as well as returning the original geometry if passed one geometry.

Reply:
Wouldn't the empty geometry be correct for no arguments?

Returning undefined sometimes means that you need to either guard the inputs or check the outputs.